src/chocolatey/infrastructure.app/runners/ConsoleApplication.cs

```diff
@@ -85,9 +88,63 @@ public void run(string[] args, ChocolateyConfiguration config, Container contain
                                 }
                             }
                         },
-                        () => command.handle_validation(config),
+                        () =>
+                        {
+                            command.handle_validation(config);
+
+                            var validationResults = new List<ValidationResult>();
+                            var validations = container.GetAllInstances<IValidation>();
+
+                            foreach (var validation in validations)
+                            {
+                                validationResults.AddRange(validation.validate(config));
+                            }
+
+                            var validationErrors = report_validation_summary(validationResults);
+
+                            if (validationErrors != 0)
+                            {
+                                // NOTE: This is intentionally left blank
+                                throw new ApplicationException("");
+                            }
+                        },
                         () => command.help_message(config));
                 });
         }
+
+        private int report_validation_summary(IList<ValidationResult> validationResults)
+        {
+            var successes = validationResults.Count(v => v.Status == ValidationStatus.Success);
+            var warnings = validationResults.Count(v => v.Status == ValidationStatus.Warning);
+            var errors = validationResults.Count(v => v.Status == ValidationStatus.Error);
+
+            this.Log().Info("{0} validations performed. {1} success(es), {2} warning(s), and {3} error(s).".format_with(
+                validationResults.Count,
+                successes,
+                warnings,
+                errors));
+
+            if (warnings != 0)
+            {
+                this.Log().Info("");
+                this.Log().Warn("Warnings:");
+                foreach (var warning in validationResults.Where(p => p.Status == ValidationStatus.Warning).or_empty_list_if_null())
+                {
+                    this.Log().Warn(" - {0}".format_with(warning.Message));
+                }
+            }
+
+            if (errors != 0)
+            {
+                this.Log().Info("");
+                this.Log().Error("Errors:");
+                foreach (var error in validationResults.Where(p => p.Status == ValidationStatus.Error).or_empty_list_if_null())
+                {
+                    this.Log().Error(" - {0}".format_with(error.Message, error.ExitCode));

```

Comment:
Your format with has 2 items, but you are only passing a single item. Perhaps you meant to do this after copy pasta? 

`this.Log().Error(" - {0} ({1})".format_with(error.Message, error.ExitCode));`

Reply:
Ah, good catch.  Original, I was outputting the exit code at this point as well, but then I noticed that this was already handled, and didn't need it anymore.

Reply:
Based on the comment below, I will add this back in though.